src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodBuilderMemberCollection.cs

```diff
@@ -422,6 +424,15 @@ private static NamedTypeSymbol ValidateBuilderType(SyntheticBoundNodeFactory F,
                     !property.IsStatic &&
                     (property.ParameterCount == 0))
                 {
+                    if (property.Type != returnType)
```

Comment:
Good question. I think the answer is yes, we should use strict equality. This is in line with Chuck's earlier comment (let's use equality rather than convertibility test) and our ability to relax it later if we decide to.